sympy/stats/compound_rv.py

```diff
@@ -27,6 +31,9 @@ def __new__(cls, s, distribution):
         if not isinstance(distribution, CompoundDistribution):
             raise ValueError("%s should be an isinstance of "
                         "CompoundDistribution"%(distribution))
+        cls.is_Finite = distribution.is_Finite
+        cls.is_Continuous = distribution.is_Continuous
+        cls.is_Discrete = distribution.is_Discrete
```

Comment:
This is a global static variable overwrite. Is it the intended behaviour?

Reply:
No, that needs to be fixed...Thanks!

Reply:
This is a bit weird. Value of static variables will be prone to change with every new object creation. I think these static variables were meant to be specific to each object rather than shared across all object.